spec.html

```diff
@@ -24320,16 +24533,287 @@ <h1>Runtime Semantics: Evaluation</h1>
+          1. If _obj_ is not an Object, throw a *TypeError* exception.
+          1. If _name_ is a property key, return ? HasProperty(_obj_, _name_).
+          1. If PrivateElementFind(_obj_, _name_) is not ~empty~, return *true*.
+          1. Return *false*.
+        1. Let _has_ be CreateBuiltinFunction(_hasClosure_, 1, *"has"*, « »).
+        1. Perform ! CreateDataPropertyOrThrow(_accessObj_, *"has"*, _has_).
+        1. Return _accessObj_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createaddinitializerfunction" type="abstract operation">
+      <h1>
+        CreateAddInitializerFunction (
+          _initializers_: a List of ECMAScript function objects,
+          _decorationState_: a Record with a field [[Finished]] (a Boolean),
+        ): a function object
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>Creates a function for a decorator's context object which can be used to add an extra initializer to the class.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _addInitializerClosure_ be a new Abstract Closure with parameters (_initializer_) that captures _initializers_ and _decorationState_ and performs the following steps when called:
+          1. If _decorationState_.[[Finished]] is *true*, throw a *TypeError* exception.
+          1. If IsCallable(_initializer_) is *false*, throw a *TypeError* exception.
+          1. Append _initializer_ to _initializers_.
+          1. Return *undefined*.
+        1. Return CreateBuiltinFunction(_addInitializerClosure_, 1, *"addInitializer"*, « »).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createdecoratorcontextobject" type="abstract operation">
+      <h1>
+        CreateDecoratorContextObject (
+          _kind_: ~class~, ~field~, ~method~, ~accessor~, ~getter~, or ~setter~,
+          _name_: a property key or a Private Name,
+          _initializers_: a List of function objects,
+          _decorationState_: a Record with a field [[Finished]] (a Boolean),
+          optional _isStatic_: a Boolean,
+        ): an Object
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It returns a context object which is passed as the second argument to decorators of the given class element.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _contextObj_ be OrdinaryObjectCreate(%Object.prototype%).
+        1. If _kind_ is ~method~, then
+          1. Let _kindStr_ be *"method"*.
+        1. Else if _kind_ is ~getter~, then
+          1. Let _kindStr_ be *"getter"*.
+        1. Else if _kind_ is ~setter~, then
+          1. Let _kindStr_ be *"setter"*.
+        1. Else if _kind_ is ~accessor~, then
+          1. Let _kindStr_ be *"accessor"*.
+        1. Else if _kind_ is ~field~, then
+          1. Let _kindStr_ be *"field"*.
+        1. Else,
+          1. Assert: _kind_ is ~class~.
+          1. Let _kindStr_ be *"class"*.
```

Comment:
aside for the editors: kind of feels like we need a `SpecValueToString()` AO or something for cases like this. i think atomics has one too.